mesonbuild/backend/ninjabackend.py

```diff
@@ -1730,10 +1734,17 @@ def generate_cython_transpile(self, target: build.BuildTarget) -> \
                         [ssrc])
                     element.add_item('ARGS', args)
                     self.add_build(element)
+                    pyx_sources.append(element)
                     # TODO: introspection?
                     cython_sources.append(output)
                 else:
                     generated_sources[ssrc] = mesonlib.File.from_built_file(gen.get_subdir(), ssrc)
+                    # Following logic in L883-900 where we determine whether to add generated source
+                    # as a header(order-only) dep to the .so compilation rule
+                    if not self.environment.is_source(ssrc) and not self.environment.is_object(ssrc) and not self.environment.is_library(ssrc) and not modules.is_module_library(ssrc):
```

Comment:
This line is very long and should really be broken up.

Reply:
Split it into four, hopefully the style is good.

The file passes flake8 FWIW, but meson's flake8 config seems to be very lax.